x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/index.tsx

```diff
@@ -56,28 +69,75 @@ const EventDetailsPanelComponent: React.FC<EventDetailsPanelProps> = ({
     skip: !expandedEvent.eventId,
   });
 
+  const [hostIsolationPanelOpen, setHostIsolationPanel] = useState(false);
+
+  const showAlertDetails = useCallback(() => {
+    setHostIsolationPanel(false);
+  }, []);
+
+  const showHostIsolationPanel = useCallback(() => {
+    setHostIsolationPanel(true);
+  }, []);
+
   const isAlert = some({ category: 'signal', field: 'signal.rule.id' }, detailsData);
 
+  const findEndpointAlert = find({ category: 'agent', field: 'agent.type' }, detailsData)?.values;
+
+  const isEndpointAlert = findEndpointAlert ? findEndpointAlert[0] === 'endpoint' : false;
+
   if (!expandedEvent?.eventId) {
     return null;
   }
 
   return isFlyoutView ? (
     <>
       <EuiFlyoutHeader hasBorder>
-        <ExpandableEventTitle isAlert={isAlert} loading={loading} />
+        {hostIsolationPanelOpen ? (
```

Comment:
nit: Same comment for large return ternary statements

Reply:
i took out the return to details link, but i feel like each block is either one component that happens to have a lot of props or a block of code that is more readable to keep inline - like the flyout footer, so probably gonna keep this mostly how it is